src/brevitas/graph/calibrate.py

```diff
@@ -191,74 +225,467 @@ def disable_act_quant_hook(self, module, inp, output):
+        disable_quant_act (bool): whether to disable activation quantization
+        disable_weight_quant (bool): whether to disable weight quantization
+        disable_bias_quant (bool): whether to disable bias quantization
+        disable_return_quant_tensor (bool): whether to disable output quantization
+        excluded_modules (list): list of submodules of modules to be excluded
+            from quantization disabling
+    """
+
+    def __init__(
+            self,
+            model: nn.Module,
+            is_training: Optional[bool] = None,
+            call_act_quantizer_impl: bool = False,
+            disable_act_quant: bool = True,
+            disable_weight_quant: bool = True,
+            disable_bias_quant: bool = True,
+            disable_return_quant_tensor: bool = True,
+            excluded_modules: Optional[List[nn.Module]] = None):
+        self.model = model
+        self.is_training = is_training if is_training is not None else model.training
+        self.prev_is_training_state = model.training
+        self.call_act_quantizer_impl = call_act_quantizer_impl
+        # Flags to disable quantization in a fined-grained manner
+        # Activations
+        self.disable_act_quant = disable_act_quant
+        self.act_quant_state = {}
+        # Weights
+        self.disable_weight_quant = disable_weight_quant
+        self.weight_quant_state = {}
+        # Bias
+        self.disable_bias_quant = disable_bias_quant
+        self.bias_quant_state = {}
+        # Return QuantTensor
+        self.disable_return_quant_tensor = disable_return_quant_tensor
+        self.return_quant_tensor_state = {}
+
+        self.excluded_modules = excluded_modules if excluded_modules is not None else []
 
-    def apply(self, model, is_training, quantization_enabled):
-        if not quantization_enabled:
-            self.disable_act_quantization(model, is_training)
-            self.disable_param_quantization(model, is_training)
-        else:
-            self.enable_act_quantization(model, is_training)
-            self.enable_param_quantization(model, is_training)
+    def __enter__(self):
+        if self.disable_act_quant:
+            self.act_quant_state = DisableEnableQuantization.disable_act_quantization(
+                model=self.model,
+                is_training=self.is_training,
+                call_act_quantizer_impl=self.call_act_quantizer_impl,
+            )
+        if self.disable_weight_quant:
+            self.weight_quant_state = DisableEnableQuantization.disable_weight_quantization(
+                model=self.model,
+                is_training=self.is_training,
+            )
+        if self.disable_bias_quant:
+            self.bias_quant_state = DisableEnableQuantization.disable_bias_quantization(
+                model=self.model,
+                is_training=self.is_training,
+            )
+        if self.disable_return_quant_tensor:
```

Comment:
This is not a self flag.
If either act quant or weight quant is disabled, then this is True otherwise you have unwanted errors, and it might be tricky for the user to understand why